src/main/java/fi/fmi/avi/archiver/config/factory/postaction/SwimRabbitMQPublisherFactory.java

```diff
@@ -126,8 +128,9 @@ public SwimRabbitMQPublisher newInstance(final Config config) {
                 })
                 .build()), publisherRef);
 
-        final SwimRabbitMQPublisher action = newSwimRabbitMQPublisher(publisher);
-        healthContributorRegistry.registerContributor(config.getId(), healthIndicator);
+        final SwimRabbitMQPublisher action = newSwimRabbitMQPublisher(publisher, publisherHealthIndicator);
+        healthContributorRegistry.registerContributor("connection", connectionHealthIndicator);
+        healthContributorRegistry.registerContributor("publisher", publisherHealthIndicator);
```

Comment:
I think we should use the `config.getId()` in the health contributor name to distinguish between possible different instances of the `SwimRabbitMQPublisher`. At simplest, just prefix with the id. Or create a method in `SwimRabbitMQConnectionHealthContributor` that creates and registers a new compound health contributor with the name of the id, containing connection and publisher health contributors provided as parameters.